docs/apm/api.asciidoc

```diff
@@ -553,3 +553,238 @@ The API returns the following:
 // More examples will go here
 
 More information on Kibana's API is available in <<api,REST API>>.
+
+////
+*******************************************************
+*******************************************************
+////
+
+[role="xpack"]
+[[rum-sourcemap-api]]
+=== RUM source map API
+
+IMPORTANT: This endpoint is only compatible with the
+{apm-server-ref}/apm-integration.html[APM integration for Elastic Agent].
+Users with a standalone APM Server should instead use the APM Server
+{apm-server-ref}/sourcemap-api.html[source map upload API].
+
+A source map allows minified files to be mapped back to original source code --
+allowing you to maintain the speed advantage of minified code,
+without losing the ability to quickly and easily debug your application.
+
+For best results, uploading source maps should become a part of your deployment procedure,
+and not something you only do when you see unhelpful errors.
+That’s because uploading source maps after errors happen won’t make old errors magically readable --
+errors must occur again for source mapping to occur.
+
+The following APIs are available:
+
+* <<rum-sourcemap-post>>
+* <<rum-sourcemap-get>>
+* <<rum-sourcemap-delete>>
+
+[float]
+[[use-sourcemap-api]]
+==== How to use APM APIs
+
+.Expand for required headers, privileges, and usage details
+[%collapsible%closed]
+======
+include::api.asciidoc[tag=using-the-APIs]
+======
+
+////
+*******************************************************
+////
+
+[[rum-sourcemap-post]]
+==== Create or update source map
+
+Create or update a source map for a specific service and version.
+
+[[rum-sourcemap-post-privs]]
+===== Privileges
+
+The user accessing this endpoint requires `All` Kibana privileges for the {beat_kib_app} feature.
+For more information, see <<kibana-privileges>>.
+
+[[apm-sourcemap-post-req]]
+===== Request
+
+`POST /api/apm/sourcemaps`
+
+[role="child_attributes"]
+[[apm-sourcemap-post-req-body]]
+===== Request body
+
+`service_name`::
+(required, string) The name of the service that the service map should apply to.
+
+`service_version`::
+(required, string) The version of the service that the service map should apply to.
+
+`bundle_filepath`::
+(required, string) The absolute path of the final bundle as used in the web application.
+
+`sourcemap`::
+(required, string) The source map. It must follow the
+https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k[Source map revision 3 proposal].
```

Comment:
@cauemarcondes, is this a string? I couldn't tell. Is there a way to upload via a file like in APM Server's API?

Reply:
Let me take a look.

Reply:
@bmorelli25 I opened a PR to add support to upload via file https://github.com/elastic/kibana/pull/105443. Now the POST API can accept both a string or a file.

Reply:
Awesome!

Reply:
Docs have been updated